baseline/tf/seq2seq/training/eager.py

```diff
@@ -13,7 +13,7 @@
 from baseline.tf.seq2seq.training.utils import to_tensors, SHUF_BUF_SZ, NUM_PREFETCH
 
 
-def loss(model, features, labels):
+def loss(model, features, labels, **kwargs):
```

Comment:
I think it would be possible to produce a functor with the right underlying implementation so that this can be done prior to loss function definition (similar to the fact we do this upfront in the PyTorch impl).  Then the loss function could be defined differently depending on if its on or not.  I guess this pushes the problem up but to me it feels weird that the optimizer ends up needing kwargs everywhere to set it up right.  Would be better IMO if we could adding the kwargs to the optimizer and loss function defs

Reply:
I link the spirit of this in that it seems to be separating the loss from the model, moving towards something that we are able to specify in the config, swap out, etc but I guess I'm not in love with the loss function being responsible for running the forward part of the model, I know it is mostly there because of how different that tagger was, but I was hoping to not to a huge refactor on how eager losses are done because I still don't know if the way we are doing it is the best way

Reply:
Updated to use a functor